pkg/actors/actors.go

```diff
@@ -370,40 +374,55 @@ func (a *actorsRuntime) haltAllActors() error {
 }
 
 func (a *actorsRuntime) deactivateActor(act *actor) error {
+	// This uses a background context as it should be unrelated from the caller's context
+	// Once the decision to deactivate an actor has been made, we must go through with it or we could have an inconsistent state
 	ctx := context.Background()
 
 	// Delete the actor from the actor table regardless of the outcome of deactivation the actor in the app
 	actorKey := act.Key()
 	a.actorsTable.Delete(actorKey)
 
-	req := invokev1.NewInvokeMethodRequest("actors/"+act.actorType+"/"+act.actorID).
-		WithActor(act.actorType, act.actorID).
-		WithHTTPExtension(http.MethodDelete, "").
-		WithContentType(invokev1.JSONContentType)
-	defer req.Close()
-
-	resp, err := a.getAppChannel(act.actorType).InvokeMethod(ctx, req, "")
+	err := a.placement.ReportActorDeactivation(ctx, act.actorType, act.actorID)
 	if err != nil {
-		diag.DefaultMonitoring.ActorDeactivationFailed(act.actorType, "invoke")
-		return err
+		return fmt.Errorf("failed to report actor deactivation for actor '%s': %w", actorKey, err)
 	}
-	defer resp.Close()
 
-	if resp.Status().GetCode() != http.StatusOK {
-		diag.DefaultMonitoring.ActorDeactivationFailed(act.actorType, "status_code_"+strconv.FormatInt(int64(resp.Status().GetCode()), 10))
-		body, _ := resp.RawDataFull()
-		return fmt.Errorf("error from actor service: %s", string(body))
+	// If it's an internal actor, we call it directly
+	_, ok := a.internalActorTypes.Get(act.actorType)
+	if ok {
+		internalAct, loaded := a.internalActors.GetAndDel(act.Key())
+		// If the actor was loaded in-memory, call DeactivateActor on it
+		if loaded {
+			err = internalAct.DeactivateActor(ctx)
+			if err != nil {
+				diag.DefaultMonitoring.ActorDeactivationFailed(act.actorType, "internal")
+				return fmt.Errorf("failed to deactivate internal actor: %w", err)
+			}
+		}
+	} else if a.appChannel != nil {
+		req := invokev1.NewInvokeMethodRequest("actors/"+act.actorType+"/"+act.actorID).
```

Comment:
Also use `daprSeparator`

Reply:
No, daprSeparator is || and here we use /